### DISCUSSION TITLE
Rewrites/redirects not working properly in production for API routes

### DISCUSSION THREAD

**Discussion by Harrisonkamau** [vercel/next.js#15864] [Help]
## Background
I want to render a purely JSON page out of the API namespace i.e. `pages/config.json.js`. 

**First problem**
[Salesforce Marketing Cloud UI](https://developer.salesforce.com/docs/atlas.en-us.noversion.mc-app-development.meta/mc-app-development/custom-activity-ui.htm) requires the presence of `endpointUrl/config.json` and `endpointUrl/config.js` rendering the same JSON configuration file.

**The solution that required duplicate implementation**
From the [Next.js docs](https://nextjs.org/docs/api-reference/data-fetching/getInitialProps), it's possible to render a JSON page in a non-api endpoint by using `getInitialProps`:
```js
// pages/config.json.js
function Component({ data }) {
  return <div>{data}</div>
}

Component.getInitialProps = async (ctx) => {
  const { res, query } = ctx;
  const configData = { name: 'Jane Doe' } // I used real config data as shown here:(https://developer.salesforce.com/docs/atlas.en-us.noversion.mc-app-development.meta/mc-app-development/custom-activity-config.htm)

  res.setHeader('Content-Type', 'application/json; charset=utf8');
  res.write(JSON.stringify(configData));
  res.end();

  return {};
};

export default Component;
```

The problem with this is that I have to duplicate the logic into another `pages/config.js.js` file to be able to access: `/config.js` endpoint.

## What I chose to do
Create a real API endpoint `pages/api/config.json.js` only (meaning I didn't have to create `pages/config.json.js` or `pages/config.js`) files and then use `redirects` (I also experimented with `rewrites` and got nowhere) to map the correct Paths.

```js
// next.config.js
async redirects() {
    return [
      {
        source: '/config.json',
        destination: '/api/config.json',
        statusCode: 308,
      },
    ];
  },
```

The above only works on the `localhost` and gives a `404` after deploying to AWS.

I deployed the app using [`serverless-components`](https://github.com/serverless-nextjs/serverless-next.js/blob/master/README.md). And my `serverless.yml`:
```yml
journeyBuilderActivities:
  component: "@sls-next/serverless-component@1.15.1"
  inputs:
    bucketName: "my-bucket-name"
```

Any thoughts or experiences?

**Comment by timneutkens** [comment]
This doesn't seem like a Next.js issue, it's probably that the serverless-components community project you're using does not support these new properties yet. Would recommend using `next build` and `next start` if you always want to use the latest features from Next.js.

**Comment by bsor-dev** [comment]
Same issue

**Comment by bornaradinovic** [comment]
same issue, using next build & start
